### PR TITLE
Dropped another small piece of LanItems

### PR DESCRIPTION
**Pull request by mchf**
Another small piece of LanItems is no longer between us.

### CODE REVIEW

File: src/lib/y2network/interface.rb [hunk 1]
```diff
@@ -132,5 +132,10 @@ def update_udev_rule
           Y2Network::UdevRule.new_bus_id_based_rename(name, hardware.busid, hardware.dev_port)
       end
     end
+
+    # @return [Boolean] true if the interface is hotplug
+    def hotplug?
+      ["usb", "pcmci"].include?(hardware["hotplug"])
```
Comment (imobachgs) [inline src/lib/y2network/interface.rb:138]: `pcmci` or `pcmcia`? I know, the original one was `pcmci`. Just curious.

File: src/include/network/lan/address.rb [hunk 2]
```diff
@@ -55,12 +55,10 @@ def initialize_network_lan_address(include_target)
     # Dialog for setting up IP address
     # @return dialog result
     def AddressDialog(builder:)
-      @builder = builder
-
       ret = Y2Network::Dialogs::EditInterface.run(builder)
 
       if ret != :back && ret != :abort
-        if LanItems.isCurrentDHCP && !LanItems.isCurrentHotplug
+        if builder.boot_protocol.dhcp? && !builder.interface.hardware.hotplug?
```
[the remark this answers is not in the document]
Comment (mchf) [reply src/include/network/lan/address.rb:61]: ups ... obviously error. will fix

File: src/lib/y2network/interface.rb (hunk 1 above)
Comment (mchf) [reply src/lib/y2network/interface.rb:138]: good question.  "pcmci" has been here longer than I work with the code. I've checked source of hwinfo / libhd. And it seems that PCMCIA is correct (as expected anyway)

See https://github.com/openSUSE/hwinfo/blob/4264ba23bc432193d0e1a522909e299cfb35b468/src/hd/hdp.c#L369

Comment (wfeldt) [reply src/lib/y2network/interface.rb]: It's `pcmcia`. And I really wonder if we should care about it at all or just remove that stuff.

Comment (mchf) [reply src/lib/y2network/interface.rb]: thanks ... so it obviously was wrong for about 10 years

[the remark this answers is not in the document]
Comment (mchf) [reply src/lib/y2network/interface.rb]: some modifications/cleanup in related code were done. Hopefully it looks better now.

Comment (wfeldt) [reply src/lib/y2network/interface.rb]: Which nicely illustrates why we might remove it.

Comment (imobachgs) [reply src/lib/y2network/interface.rb]: So it remained unused for about 10 years :smile: Actually, we removed another PCMCIA-related feature (in the autoinstallation rules) back in 2016. It never worked for SLE 12. So I agree with @wfeldt here.

**Review by imobachgs — APPROVED** [review]

**Pull request merged**